website/docs/pipelines/promoting-applications.mdx

```diff
@@ -185,13 +202,97 @@ type: Opaque
+
 :::caution
-The Secret needs to reside in the same Namespace as the Pipeline resource on the management cluster.
+
+Ensure you adopt the recommendations in this section to reduce the risks associated with the secrets involved in pull requests.
 :::
 
+1. **Do not use user accounts**: use service or bot accounts. User context from the credentials would be used to do any git provider operation and for
+security and auditing perspective you don't want your user context be used in machine operations.
+
+<details><summary>Expand to see example</summary>
+
+![using bot account](img/bot-account.png)
+</details>
+
+2. **Restrict access to the secret**: the promotion credentials needs to reside in the same Namespace as the Pipeline resource on the management cluster. Restrict
+via RBAC that only `pipeline-controller` service account is able to read it.
+
+<details><summary>Expand to see example</summary>
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: read-app-promotion-credentials
+  namespace: app
+rules:
+  - apiGroups:
+      - ""
+    resourceNames:
+      - "app-promotion-credentials"
+    resources:
+      - "secrets"
+    verbs:
+      - "get"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: pipeline-controller-read-app-promotion-credentials
+  namespace: app
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: read-app-promotion-credentials
+subjects:
+  - kind: ServiceAccount
+    name: pipeline-controller
+    namespace: pipeline-system
+```
+</details>
+
+3. **Do not use long-live tokens**: set an expiration date and rotate them according to your security policy.
+
+<details><summary>Expand to see example</summary>
+
+![create token witih expiration](img/create-token-with-expiration.png)
+</details>
+
+4. **Honour the least privilege principle**: avoid having high privilege tokens. Restrict the token to your just your repo and to just the operations required.
+
+<details><summary>Expand to see example</summary>
+
+For example, if the case of GitHub, use [fine-grained tokens](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/) to only
+allow access to the single repo that your configuration manifests exist.
+
+![create token least priviledge](img/fine-grained-token.png)
+ </details>
+
+5. **Review active access tokens on a regular basis**: to ensure that only the ones that are required are present at all times.
+
+<details><summary>Expand to see example</summary>
+
+For example, using github and fine-grained tokens you [could do so](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/#approving-and-auditing-personal-access-tokens).
+
+![review tokens](img/manage-fine-grained.png)
+</details>
+
+6. **Reduce number of secrets to maintain**: Whenever ends in a more secure setup, favor git over https with tokens instead of ssh and deploy keys. Given that a token
```

Comment:
```suggestion
6. **Reduce number of secrets to maintain**: Whenever ends in a more secure setup, favour git over https with tokens instead of ssh and deploy keys. Given that a token
```

You use _honour_ here https://github.com/weaveworks/weave-gitops/pull/3426/files#diff-b64d18b8c0ecf5a67a9bed187be99a4ed49057ecba3d55bb03de2b9a681165e8R269 these should be consistent.